rust/datafusion/src/physical_plan/hash_join.rs

```diff
@@ -575,6 +542,199 @@ fn build_join_indexes(
         }
     }
 }
+use core::hash::BuildHasher;
+
+/// `Hasher` that returns the same `u64` value as a hash, to avoid re-hashing
+/// it when inserting/indexing or regrowing the `HashMap`
+struct IdHasher {
+    hash: u64,
```

Comment:
We just return the key value here, so we don't "rehash" when growing or indexing the `HashMap`